data/account/participationRegistry.go

```diff
@@ -691,42 +767,14 @@ func (db *participationDB) Flush() error {
 	db.mutex.Lock()
 	defer db.mutex.Unlock()
```

Comment:
Is the mutex still required? Seems like this would cause a deadlock when the writeThread attempts to Lock

Reply:
the flushDone condition variable temporarily releases its lock (`&db.mutex`) inside `.Wait()`; that's the receiving end of `db.flushDone.Broadcast()` in `writeThread()`

Reply:
Ahh, that's pretty cool. I found a couple of cases that I thought were race conditions but they seem to all be covered by the pattern. (multiple calls to flush, the window between sending the flush event/calling wait)

One potential issue I see... 

If an error is generated with multiple Flush() calls on the queue, we guarantee that only the first Flush returns the error. This might happen if we use Flush in the REST API and are inserting keys back-to-back.

I think we can solve that by counting the number of pending flushes and only clearing `asyncErr` when changing the pending account from 0 -> 1.